test/e2e/kind_test.go

```diff
@@ -302,10 +304,61 @@ func (s *kindSuite) TestKindRun() {
 		}, 600*time.Second, 30*time.Second, "could not validate kubernetes_state_core check with api client")
 	})
 
+	s.T().Run("Logs collection works", func(t *testing.T) {
+		// Update DDA
+		ddaConfigPath, err = getAbsPath(filepath.Join(manifestsPath, "datadog-agent-logs.yaml"))
+		assert.NoError(t, err)
+
+		k8s.KubectlApply(t, kubectlOptions, ddaConfigPath)
+		verifyAgentPods(t, kubectlOptions, nodeAgentSelector+",agent.datadoghq.com/e2e-test=datadog-agent-logs")
+		verifyNumPodsForSelector(t, kubectlOptions, 1, clusterAgentSelector+",agent.datadoghq.com/e2e-test=datadog-agent-logs")
+
+		// Verify logs collection on agent pod
+		s.EventuallyWithTf(func(c *assert.CollectT) {
+			agentPods, err := k8s.ListPodsE(t, kubectlOptions, v1.ListOptions{
+				LabelSelector: nodeAgentSelector + ",agent.datadoghq.com/e2e-test=datadog-agent-logs",
+			})
+			assert.NoError(c, err)
+
+			for _, pod := range agentPods {
+				k8s.WaitUntilPodAvailable(t, kubectlOptions, pod.Name, 9, 15*time.Second)
+
+				output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", "-it", pod.Name, "--", "agent", "status", "logs agent", "-j")
+				assert.NoError(c, err)
+
+				verifyAgentPodLogs(c, output)
+			}
+		}, 900*time.Second, 30*time.Second, "could not validate log check on agent pod")
+
+		s.EventuallyWithTf(func(c *assert.CollectT) {
+			verifyAPILogs(s, c)
+		}, 600*time.Second, 30*time.Second, "could not valid logs collection with api client")
+
+	})
+
 	s.T().Run("Cleanup DDA", func(t *testing.T) {
 		deleteDda(t, kubectlOptions, ddaConfigPath)
 	})
 }
+
+func verifyAgentPodLogs(c *assert.CollectT, collectorOutput string) {
+	var agentLogs []interface{}
+	logsJson := parseCollectorJson(collectorOutput)
+
+	if logsJson != nil {
+		agentLogs = logsJson["logsStats"].(map[string]interface{})["integrations"].([]interface{})
+		for _, log := range agentLogs {
+			if integration, ok := log.(map[string]interface{})["sources"].([]interface{})[0].(map[string]interface{}); ok {
+				message, exists := integration["messages"].([]interface{})[0].(string)
+				assert.True(c, exists)
+				assert.Contains(c, message, "files tailed", "expected message to contain 'files tailed'")
```

Comment:
Just wondering, are there any attributes in the `logsStats` json that show a count of log files being tailed? Is there any chance that the log `message` reads `0 files tailed` or something like that? That could give a false positive

Reply:
There aren't, but all of the `message` contents are `1 files tailed out of 1 files matching` in the e2e test. When I was testing it locally sometimes it would vary which is why I just put `"files tailed"` in the test.